code/game/objects/effects/decals/Cleanable/aliens.dm

```diff
@@ -21,18 +21,21 @@
 	return ..()
 
 /obj/effect/decal/cleanable/xenoblood/xgibs/proc/streak(list/directions)
-	spawn (0)
-		var/direction = pick(directions)
-		for (var/i = 0, i < pick(1, 200; 2, 150; 3, 50; 4), i++)
-			sleep(3)
-			if (i > 0)
-				var/obj/effect/decal/cleanable/xenoblood/b = new /obj/effect/decal/cleanable/xenoblood/xsplatter(src.loc)
-				for(var/datum/disease/D in src.viruses)
-					var/datum/disease/ND = D.Copy(1)
-					b.viruses += ND
-					ND.holder = b
-			if (step_to(src, get_step(src, direction), 0))
-				break
+	set waitfor = 0
+	var/direction = pick(directions)
+	for(var/i = 0, i < pick(1, 200; 2, 150; 3, 50), i++)
+		sleep(2)
+		if (i > 0)
+			var/obj/effect/decal/cleanable/xenoblood/b = new /obj/effect/decal/cleanable/xenoblood/xsplatter(src.loc)
+			for(var/datum/disease/D in src.viruses)
+				var/datum/disease/ND = D.Copy(1)
+				b.viruses += ND
+				ND.holder = b
+		if (!step_to(src, get_step(src, direction), 0))
+			break
+
+/obj/effect/decal/cleanable/xenoblood/xgibs/replace_decal(obj/effect/decal/cleanable/C)
+	return //gibs can stack, so gibspawner can work properly

```

Comment:
if you create a cleanable decal on top of another, the two are merged, this is so a floor tile doesn't have a dozen blood effects on it. But this prevents gibspawner from working (spawns all gibs first, then make them slide all together at the same time).